src/test-utils.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { FC, ReactElement, ReactNode } from 'react'
+import { render, RenderOptions } from '@testing-library/react'
+import ThemeProvider from 'theme'
+import store from 'state'
+import { Provider } from 'react-redux'
+import { I18nProvider } from '@lingui/react'
+import { i18n } from '@lingui/core'
+import getLibrary from 'utils/getLibrary'
+import { createWeb3ReactRoot, Web3ReactProvider } from '@web3-react/core'
+import { NetworkContextName } from 'constants/misc'
+import { ThemedGlobalStyle } from 'theme'
+import { HashRouter } from 'react-router-dom'
+
+const Web3ProviderNetwork = createWeb3ReactRoot(NetworkContextName)
+
+const WithProviders: FC = ({ children }: { children?: ReactNode }) => {
+  return (
+    <Provider store={store}>
```

Comment:
you don't need all this stuff for most component tests, including the ones included in this pr afaict. is this a good idea? seems like you're going to end up bundling and rendering half the app, plus making actual network requests in every unit test rather than just what's needed

Reply:
Those were needed for some other components i'm testing locally but I agree it's not ideal. I'll remove them from here for now and consider switching to `jest.mock`